src/dataframe/dataframe.jl

```diff
@@ -777,39 +777,144 @@ end
 deletecols!(df::DataFrame, c::Int) = deletecols!(df, [c])
 deletecols!(df::DataFrame, c::Any) = deletecols!(df, index(df)[c])
 
-function deleterows!(df::DataFrame, ind::Union{Integer, UnitRange{Int}})
+function deleterows!(df::DataFrame, ind)
+    # we require ind to be stored and unique like in Base
+    keep = deleteat!(collect(axes(df, 1)), ind)
```

Comment:
I'd prefer removing in-place, that's one of the interests of having `deleterows!`. I guess we could also provide `deleterows` which would make a copy.

In terms of implementation, it will probably be more efficient to have a method for `AbstractVector{Bool}` which will call `findall`, to avoid computing that for each column. BTW, that's probably something we should also do for `getindex`.